rspec-core.gemspec

```diff
@@ -44,7 +44,10 @@ Gem::Specification.new do |s|
     s.add_runtime_dependency "rspec-support", "~> #{RSpec::Core::Version::STRING.split('.')[0..1].concat(['3']).join('.')}"
   end
 
-  s.add_development_dependency "cucumber", "~> 1.3"
+  if RUBY_VERSION >= '2.5'
+    s.add_development_dependency "cucumber", "~> 7.0.0"
```

Comment:
I would propose versioning the cucumber dependency as ">= 3.2", "< 8.0", and putting any extra logic needed in the Gemfile.

Reply:
It should just be worked out based on the diff-lcs requirement and ruby version no?

Reply:
Yes, I think those are basically the two inputs that would determine which cucumber versions will work.

Reply:
Should also be `!= 4.x.x` (long list of  4.1.0 4.0.1 4.0.0 4.0.0.rc.6 4.0.0.rc.5 4.0.0.rc.4 4.0.0.rc.3 4.0.0.rc.2 4.0.0.rc.1)

Reply:
Ah, my bad, 4.x was incompatible due to `diff-lcs`, so we're fine.

I'll update https://github.com/rspec/rspec-expectations/pull/1320 to work around `ok?` error on 3.2 somehow.

Reply:
Thanks for the suggestion, worked perfectly.

Fun fact: on Rubies 2.3, 2.4 and JRuby 9.1, Bundler installs `cucumber` 4.0.0.

Reply:
> Fun fact: on Rubies 2.3, 2.4 and JRuby 9.1, Bundler installs cucumber 4.0.0.

Oh, that's funny: Cucumber 4.0.1 disallows diff-lcs 1.4.4, but version 4.0.0 allows it. There must be some problem with diff-lcs 1.4 but it clearly isn't relevant here.

Reply:
[4.0.1's changelog mentions `diff-lcs`](https://github.com/cucumber/cucumber-ruby/blob/main/CHANGELOG.md#401)'s breaking changes.

I've added `"!= 4.0.0"` to the specification to be on the safe side.
2.3, 2.4 and JRuby 9.1 builds will use Cucumber 3.2, while others will take the latest 7.x.

It's remarkable that Cucumber mentions to [drop support for JRuby in 4.0 rc](https://github.com/cucumber/cucumber-ruby/commit/2ead65ac3ff5206b938e46958a17659af36341d0).